pkg/build/controller/strategy/util.go

```diff
@@ -90,3 +65,45 @@ func setupBuildEnv(build *buildapi.Build, pod *kapi.Pod) error {
 	}
 	return nil
 }
+
+// setupDockerSecrets mounts Docker Registry secrets into Pod running the build,
+// allowing Docker to authenticate against private registries or Docker Hub.
+func setupDockerSecrets(pod *kapi.Pod, pushSecret string) {
+	if len(pushSecret) == 0 {
+		return
+	}
+
+	volume := kapi.Volume{
+		Name: pushSecret,
+		VolumeSource: kapi.VolumeSource{
+			Secret: &kapi.SecretVolumeSource{
+				Target: kapi.ObjectReference{
+					Kind: "Secret",
+					Name: pushSecret,
+					// TODO: The namespace should not be required here.
+					//       See: https://github.com/GoogleCloudPlatform/kubernetes/pull/5807
+					Namespace: pod.Namespace,
+				},
+			},
+		},
+	}
+	volumeMount := kapi.VolumeMount{
+		Name:      pushSecret,
+		MountPath: dockerSecretMountPath,
+		ReadOnly:  true,
+	}
+
+	glog.V(3).Infof("Adding %s secret to build Pod %s", pushSecret, pod.Name)
+	pod.Spec.Volumes = append(pod.Spec.Volumes, volume)
+	pod.Spec.Containers[0].VolumeMounts = append(pod.Spec.Containers[0].VolumeMounts, volumeMount)
+
+	// TODO: The push && pull secrets are now the same, but the pull secret should
+	//       be delivered by Service Account in future.
+	// TODO: The Secret.Name must match with one if the Secret.Data[] keys in
+	//       in order to provide the full path to dockercfg. If it does not, then
+	//       the builder fail to find the dockercfg
+	pod.Spec.Containers[0].Env = append(pod.Spec.Containers[0].Env, []kapi.EnvVar{
+		{Name: "PULL_DOCKERCFG_PATH", Value: filepath.Join(volumeMount.MountPath, pushSecret)},
```

Comment:
1) more discoverable
2) easier to change the path if we decide to do so later w/o breaking anything/anyone (eg custom builder images)